edx/edx-enterprise
# WL-1242: Specify Data Sharing Consent wording on a per Enterprise basis

**Description:** This PR has changes to specify Data Sharing Consent wording on a per Enterprise basis.

**JIRA:** https://openedx.atlassian.net/browse/WL-1242

**Dependencies:** None

**Merge deadline:** None

**Testing instructions:**

1. Install edx-enterprise in develop mode and checkout ziafazal/WL-1512 branch
2. Enable edx-enterprise in edx-platform via feature flag
3. Run migrations
`./manage lms migrate --settings=devstack`
4. Add New enterprise customer via django admin
Home › Enterprise › Enterprise Customers
Make sure `Enable data sharing consent` checkbox is checked and `Enforce data sharing consent` is set to `At Enrollment`
5. Create a Data sharing consent page for newly created enterprise customer via django admin
Home › Enterprise Consent › Data sharing consent pages › Add data sharing consent page
6. After saving Data sharing consent page open it again in edit mode. Click on the `Preview (course)` button or `Preview (program)` on top right to preview your newly created data sharing page in course or program context.
7. Repeat step 4, 5 and 6 to create Data sharing consent page for an other customer and preview it.

**Sandbox Testing instructions:**
The latest version of this branch is installed at:

- https://bill-filler.sandbox.edx.org
- edx-themes has been updated to remove the MIT overrides for DSC files
- 3 enterprise customers are configured (EdXEnterprise, Boeing, Haliburton), this can be seen via django-admin https://bill-filler.sandbox.edx.org/admin/enterprise/enterprisecustomer/
- Boeing and Haliburton are each associated with site https://mitxpro-bill-filler.sandbox.edx.org. EdxEnterprise is associated with https://bill-filler.sandbox.edx.org
- Custom Data Sharing Consent pages have been setup for Boeing and Haliburton https://bill-filler.sandbox.edx.org/admin/consent/datasharingconsentpage/

1. Add a separate learner for each enterprise via Enterprise->Enterprise Customer->Manage Learners
2. For EdxEnterprise, manually enroll learner edx@test.com in course course-v1:edX+DemoX+Demo_Course
3. For Boeing, manually enroll learner boeing@test.com in course course-v1:MITxPRO+WL_E2E+2018
4. For Halliburton, manually enroll learner hal@test.com in course course-v1:MITxPRO+WL_E2E+2018
5. For EdXEnterprise, go to bill-filler.sandbox.edx.org and create an account for edx@test.com. Then go to dashboard and try to access course-v1:edX+DemoX+Demo_Course. You should see the default Data Sharing Consent form
6. For Boeing, got to mitxpro-bill-filler.sandbox.edx.org, and create an account for boeing@test.com
7. Then go to dashboard and try to access course-v1:MITxPRO+WL_E2E+2018. You should see customized Data Sharing Consent form configured for Boeing.
8. For Halliburton, got to mitxpro-bill-filler.sandbox.edx.org, and create an account for hal@test.com
9. Then go to dashboard and try to access course-v1:MITxPRO+WL_E2E+2018. You should see customized Data Sharing Consent form configured for Halliburton.

**Merge checklist:**


**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
